helusers/authz.py

```diff
@@ -42,9 +43,24 @@ def has_api_scope_with_prefix(self, prefix):
 
     @cached_property
     def _authorized_api_scopes(self):
-        api_scopes = self.data.get(self.settings.API_AUTHORIZATION_FIELD)
-        return (set(api_scopes)
-                if is_list_of_non_empty_strings(api_scopes) else None)
+        authorization_fields = self.settings.API_AUTHORIZATION_FIELD
+        if isinstance(authorization_fields, str):
+            authorization_fields = [authorization_fields]
+
+        collected_api_scopes = []
+        for authorization_field in authorization_fields:
+            api_scopes = self.data.get(authorization_field)
+            if api_scopes is None and "." in authorization_field:
+                api_scopes = flatten_list(
+                    get_nested_from_dict(self.data, authorization_field)
+                )
+                api_scopes = list(filter(None, api_scopes))
+            collected_api_scopes.extend(api_scopes)
+
+        if not is_list_of_non_empty_strings(collected_api_scopes):
+            return None
+
+        return set(collected_api_scopes)
```

Comment:
The `UserAuthorization` and `JWT` classes seem to contain similar code for the scopes validation. Could that code be refactored somehow to not be repeated in two places?

Reply:
I extracted the api scope gathering to it's own function. Left the scope checks as-is as they are so small and simple. 

Do you think the `UserAuthorization.has_api_scope_with_prefix` and `JWT.validate_api_scope` should have a common implementation?

Reply:
There are some similarities between those two, but also differences. Perhaps they are good as they are.